src/Microsoft.OData.Core/ODataBatchWriter.cs

```diff
@@ -313,81 +313,34 @@ public Task<ODataBatchOperationResponseMessage> CreateOperationResponseMessageAs
 #endif
 
         /// <summary>Flushes the write buffer to the underlying stream.</summary>
-        public void Flush()
-        {
-            this.VerifyCanFlush(true);
-
-            // make sure we switch to state FatalExceptionThrown if an exception is thrown during flushing.
-            try
-            {
-                this.rawOutputContext.Flush();
-            }
-            catch
-            {
-                this.SetState(BatchWriterState.Error);
-                throw;
-            }
-        }
+        public abstract void Flush();
 
 #if ODATALIB_ASYNC
         /// <summary>Flushes the write buffer to the underlying stream asynchronously.</summary>
         /// <returns>A task instance that represents the asynchronous operation.</returns>
-        public Task FlushAsync()
-        {
-            this.VerifyCanFlush(false);
-
-            // make sure we switch to state FatalExceptionThrown if an exception is thrown during flushing.
-            return this.rawOutputContext.FlushAsync().FollowOnFaultWith(t => this.SetState(BatchWriterState.Error));
-        }

```

Comment:
As above; you should be able to keep this public implementation in the base class and remove from derived classes by referencing this.outputContext, rather than this.rawOutputContext.